src/history/hs_rec.c

```diff
@@ -912,6 +912,48 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
         if (__wt_txn_tw_stop_visible_all(session, twp))
             continue;
 
+        /*
+         * The below condition is to handle scenario where the data store and the
+         * history store has same records and ignore.
```

Comment:
The below example illustrates a case that the data store and the history store may contain the same value. In this case, skipping inserting the same value to the history store again.